### ISSUE TITLE
Circular Pointer X-pessimism Problem

### ISSUE THREAD

**Issue by gaozihou** [bespoke-silicon-group/basejump_stl#491]
https://github.com/bespoke-silicon-group/basejump_stl/blob/master/bsg_misc/bsg_circular_ptr.v#L29-L32

Should explicitly specify 
// synopsys sync_set_reset "reset_i"
to prevent X-pessimism problem

**Comment by dpetrisko** [comment]
Best practice might be to use a bsg_dff_reset?

**Comment by gaozihou** [comment]
> Best practice might be to use a bsg_dff_reset?

Yeah I guess so, although bsg_dff_reset does not have //synopsys sync_set_reset as well.. Should work without ungrouping.

**Comment by dpetrisko** [comment]
Oh we could put the pragma in there
